crates/ruff/src/source_kind.rs

```diff
@@ -1,8 +1,9 @@
+use crate::autofix::source_map::SourceMap;
 use crate::jupyter::Notebook;
 
 #[derive(Clone, Debug, PartialEq, is_macro::Is)]
 pub enum SourceKind {
-    Python,
+    Python(String),
```

Comment:
It would be nice to store the `PySourceType` on the `SourceKind` as well (making it a `Program` or `SourceFile` entity). However, it is a bit awkward because `PySourceType` has a `Jupyter` variant, but that variant should only be used to `Jupyter`...